app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,66 @@
+class PaymentsController < ApplicationController
+  before_action :authenticate_user!, except: [:gift, :create_gift]
+  before_action :skip_signed_up_users, except: [:gift, :create_gift]
+  before_action :set_price, except: :create
+
+  def new
+    render layout: 'sign_up_flow'
+  end
+
+  def gift
+    @gift = Gift.new
+    render layout: 'sign_up_flow'
+  end
+
+  def create_gift
+    @gift = Gift.new(gift_params)
+    @gift.build_coupon
+    @payment = Payment.new(payment_params)
+    @payment.email = @gift.your_email
+    @payment.coupon = found_coupon?
+    @payment.gift = true
+    if @gift.valid? && @payment.valid?
+      ActiveRecord::Base.transaction do
+        @gift.save
+        @payment.save(validate: false)
```

Comment:
really need to refactor this. the issue is that validate calls stripe for a token. save and valid? both call validations. perhaps i can memoize the value.